browser/tor/brave_tor_client_updater_delegate.h

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_TOR_BRAVE_TOR_CLIENT_UPDATER_DELEGATE_H_
+#define BRAVE_BROWSER_TOR_BRAVE_TOR_CLIENT_UPDATER_DELEGATE_H_
+
+#include "base/files/file_path.h"
+#include "brave/components/tor/brave_tor_client_updater.h"
+
+namespace tor {
+
+class BraveTorClientUpdaterDelegate : public BraveTorClientUpdater::Delegate {
+ public:
+  explicit BraveTorClientUpdaterDelegate(const base::FilePath& user_data_dir);
+  ~BraveTorClientUpdaterDelegate() override;
+
+  // BraveTorClientUpdater::Delegate
```

Comment:
I think tor component can do cleanup and check its disabled state by passing `user_prefs` and `user data dir` to component instead of using Delegate. WDYT?

Reply:
I also think this delegate isn't needed.

Reply:
addressed in 2025088c945de7d44c240623355c7ececb4d6786